### PR TITLE
skeleton for win modal

### PR DESCRIPTION
**Pull request by etha7**
### Motivation
---
We probably want a model to display when users win.
### Changes
---
Add a simple version of the modal, without win logic

### Tests
---
image

### Picture of Cat (Not Optional)
---
image

### CODE REVIEW

File: frontend/src/components/Modal.tsx [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+
+const ModalBox = styled.div`
+  background: whitesmoke;
+  padding: 1em;
+  border-radius: 1em;
+`
+
+export type ModalProps = { children: string };
+export const Modal = ({children,}: ModalProps) => {
+  let [showModal, setShowModal] = useState(true);
```
Comment (normanjli) [inline frontend/src/components/Modal.tsx:25]: Use const here, it helps prevent accidental mutations to react controlled stated objects.

File: frontend/src/components/Modal.tsx [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+
+  const closeModal = () => {
+    setShowModal(false)
+  }
+  const modal : JSX.Element | null = showModal ? 
+                  <ModalBackground>
+                    <ModalBox onClick={closeModal}> You win!</ModalBox> 
+                  </ModalBackground>
```
Comment (normanjli) [inline frontend/src/components/Modal.tsx:33]: Could add a small x button and then add some listeners to the background to improve user experience. The listeners would listen for escape key and click on the modal to close it 

Something to think about for the next ticket.

**Review by normanjli — APPROVED** [review]
Looks good, added some comments for future thought.

File: frontend/src/components/Modal.tsx (hunk 1 above)
Comment (etha7) [reply frontend/src/components/Modal.tsx]: professional let -> const linter

good idea

File: frontend/src/components/Modal.tsx (hunk 2 above)
Comment (etha7) [reply frontend/src/components/Modal.tsx:33]: good idea, especially the escape key.

Comment (JoshIce) [reply frontend/src/components/Modal.tsx:33]: I personally really like when modals let you click anywhere on the screen outside of the modal to close them. Is this default behavior here or something we'd have to implement?

Comment (etha7) [reply frontend/src/components/Modal.tsx:33]: > 

will do this

Comment (etha7) [reply frontend/src/components/Modal.tsx:39]: done

**Review by JoshIce — APPROVED** [review]

**Pull request merged**